### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by davidd-da**
* Use explicit `JDK_JAVA_OPTIONS` in Docker Compose (picked up in Java 9+) instead of using custom `-XX:+CrashOnOutOfMemoryError` + `JAVA_OPTS` in Daml services docker image entrypoint
* Avoid overwriting docker cache layer using the same file name in `ADD` statement
* Bump:
  * Postgres to 11.9 (oldest supported version)
  * Prometheus to 2.37.6 (LTS)
  * Grafana 9.3.11
  * Loki + Promtail 2.7.5

### CODE REVIEW

File: daml-service/entrypoint.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 [[ -n ${VERBOSE:-} ]] && set -x
 
-# Log everything (stdout and stderr) to both log file and terminal
-exec > >(tee /var/log/promtail/"${DAML_SERVICE}".log) 2>&1
+# Bash trick - Log everything (stdout and stderr) to both log file and terminal
+exec > >(tee -a "/var/log/promtail/${DAML_SERVICE}.log") 2>&1
```
Comment (davidd-da) [inline daml-service/entrypoint.sh:7]: Do not overwrite the file on potential restart, just append

File: daml-service/entrypoint.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
...
 
-eval "exec java -XX:+CrashOnOutOfMemoryError ${JAVA_OPTS:-} -jar /opt/service.jar \"\$@\""
+eval "exec java -jar /opt/${DAML_SERVICE}.jar \"\$@\""
```
Comment (nicu-da) [inline daml-service/entrypoint.sh:9]: Why remove the `CrashOnOutOfMemoryError` though? How is that related to switching to `JDK_JAVA_OPTIONS`?

Comment (hrischuk-da) [reply daml-service/entrypoint.sh:9]: I don't think we need that for the demo.  I removed it because it may have been a premature kill for the Canton container.  I did leave a comment that it can be added by the user.

Comment (davidd-da) [reply daml-service/entrypoint.sh]: Moved `-XX:+CrashOnOutOfMemoryError` to `JDK_JAVA_OPTIONS` in docker compose files so its explicit to users

File: daml-service/entrypoint.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 [[ -n ${VERBOSE:-} ]] && set -x
 
-# Log everything (stdout and stderr) to both log file and terminal
-exec > >(tee /var/log/promtail/"${DAML_SERVICE}".log) 2>&1
+# Bash trick - Log everything (stdout and stderr) to both log file and terminal
+exec > >(tee -a "/var/log/promtail/${DAML_SERVICE}.log") 2>&1
 
-eval "exec java -XX:+CrashOnOutOfMemoryError ${JAVA_OPTS:-} -jar /opt/service.jar \"\$@\""
+exec java -jar "/opt/${DAML_SERVICE}.jar" "${@}"
```
Comment (davidd-da) [inline daml-service/entrypoint.sh:9]: Simplify final `exec` statement (no need for eval and escaping anymore)

**Review by nicu-da — APPROVED** [review]
